### PR TITLE
docs(addons-toolbar): improve globalArgs usage guide

### PR DESCRIPTION
**Pull request by flo-sch**
Issue:

## What I did

Edited `@addons-toolbar` README doc based on discussion in #10579

## How to test

- Is this testable with Jest or Chromatic screenshots? no
- Does this need a new example in the kitchen sink apps? no
- Does this need an update to the documentation? no

### CODE REVIEW

File: addons/toolbars/README.md [hunk 1]
```diff
@@ -63,14 +72,25 @@ You should see a dropdown in your toolbar with options `light` and `dark`.
...
 
-```js
-const styledComponentsThemeDecorator = (storyFn, { globalArgs: { theme } }) => (
-  <ThemeProvider {...getTheme(theme)}>{storyFn()}</ThemeProvider>
-);
+```ts
+import { ThemeProvider } from 'styled-components';
+import { StoryContext, StoryGetter, StoryWrapper } from '@storybook/addons';
+
+const withThemeProvider: StoryWrapper = (Story: StoryGetter, context: StoryContext) => {
```
Comment (flo-sch) [inline addons/toolbars/README.md:81]: ping @shilman: I am not certain what should be encouraged in the documentation: write decorators as functions or use the `makeDecorator()` api, so this was just a first guess here, I can update if needed

File: addons/toolbars/README.md [hunk 2]
```diff
@@ -63,14 +72,25 @@ You should see a dropdown in your toolbar with options `light` and `dark`.
...
+  // context.globalArgs.theme here will be either 'light' or 'dark'
+  // getTheme being a function retrieving the actual theme object from that value
+  const theme = getTheme(context.globalArgs.theme);
+
+  return (
+    <ThemeProvider theme={theme}>
+      <Story {...context} />
```
Comment (flo-sch) [inline addons/toolbars/README.md:88]: Leaving that spread for now since it seems to be needed with TypeScript (see #10579)

**Comment by flo-sch** [comment]
Also, those examples seems pretty tied to React, is that fine?
I use only Storybook with React myself.
I'm not sure if the documentation should be framework agnostic?

File: addons/toolbars/README.md (hunk 1 above)
Comment (shilman) [reply addons/toolbars/README.md:81]: This is perfect.  makeDecorator generally shouldn’t be used in user code

File: addons/toolbars/README.md (hunk 2 above)
Comment (shilman) [reply addons/toolbars/README.md:88]: Ok, I'm not happy with the typings here but I think that's a separate issue. cc @tmeasday @wKich

**Review by shilman — APPROVED** [review]
Thanks @flo-sch 💯much appreciated!!!

File: addons/toolbars/README.md (hunk 2 above)
Comment (flo-sch) [reply addons/toolbars/README.md:88]: I would be interested to know how this can be avoided as well then :)

**Comment by shilman** [comment]
I hope we can overhaul the documentation in the future to make it properly mutli-framework. For now, React gets an order of magnitude more usage than any other framework, so I believe even in the updated docs we should lead with React

File: addons/toolbars/README.md (hunk 2 above)
Comment (flo-sch) [reply addons/toolbars/README.md:88]: Actually, it seems that `StoryWrapper` as a return type does not work in the end.
 
Following the types warnings, I only come with `DecoratorFunction<any>`, would really love some help on that 😅

Comment (wKich) [reply addons/toolbars/README.md:88]: I think, we could try to solve typings issue as part of the integration common types https://github.com/storybookjs/csf

**Pull request merged**